packages/wallet/backend/tests/transaction/service.test.ts

```diff
@@ -0,0 +1,299 @@
+import { PaymentPointerService } from '@/paymentPointer/service'
+import { faker } from '@faker-js/faker'
+import { User } from '@/user/model'
+import { createUser } from '@/tests/helpers'
+
+describe('Transaction Controller', (): void => {
+  let bindings: Container<Bindings>
+  let appContainer: TestApp
+  let knex: Knex
+  let authService: AuthService
+  let accountService: AccountService
+  let paymentPointerService: PaymentPointerService
+  let transactionService: TransactionService
+  let req: MockRequest<Request>
+  let res: MockResponse<Response>
+  let userId: string
+
+  const args = mockLogInRequest().body
+
+  // "dependency" = ALL foreign keys (paymentPointers, account, user...)
+  const prepareTransactionDependencies = async () => {
+    const accountServiceDepsMocked = {
+      rafiki: {
+        getAssetById: (id: unknown) =>
+          mockedListAssets.find((asset) => asset.id === id)
+      },
+      rapyd: {
+        issueVirtualAccount: () => ({
+          status: {
+            status: 'SUCCESS'
+          },
+          data: {
+            id: 'mocked'
+          }
+        }),
+        simulateBankTransferToWallet: () => ({
+          status: {
+            status: 'SUCCESS'
+          },
+          data: {
+            transactions: [
+              {
+                id: 'mocked'
+              }
+            ]
+          }
+        }),
+        getAccountsBalance: () => ({
+          data: [
+            {
+              currency: mockedListAssets[0].code,
+              balance: 777
+            }
+          ] as Partial<RapydAccountBalance>
+        })
+      }
+    }
+    const ppServiceDepsMocked = {
+      accountService: await bindings.resolve('accountService'),
+      env: await bindings.resolve('env'),
+      rafikiClient: {
+        createRafikiPaymentPointer: () => ({
+          id: faker.string.uuid(),
+          url: faker.string.alpha(10)
+        })
+      }
+    }
+    Reflect.set(accountService, 'deps', accountServiceDepsMocked)
+    Reflect.set(paymentPointerService, 'deps', ppServiceDepsMocked)
+
+    const account = await accountService.createAccount({
+      userId,
+      name: faker.string.alpha(10),
+      assetId: mockedListAssets[0].id
+    })
+    const paymentPointer = await paymentPointerService.create(
+      userId,
+      account.id,
+      faker.string.alpha(10),
+      faker.string.alpha(10)
+    )
+
+    return {
+      account,
+      paymentPointer
+    }
+  }
+
+  beforeAll(async (): Promise<void> => {
+    bindings = createContainer(env)
+    appContainer = await createApp(bindings)
+    knex = appContainer.knex
+    authService = await bindings.resolve('authService')
+    accountService = await bindings.resolve('accountService')
+    paymentPointerService = await bindings.resolve('paymentPointerService')
+    transactionService = await bindings.resolve('transactionService')
+  })
+
+  beforeEach(async (): Promise<void> => {
+    res = createResponse()
```

Comment:
as the service is tested, there is no use for req, res as the user id and other details are passed as params